crates/re_viewer/src/ui/add_space_view_or_container_modal.rs

```diff
@@ -0,0 +1,142 @@
+
+use itertools::Itertools;
+
+use re_log_types::{EntityPath, EntityPathFilter};
+use re_space_view::DataQueryBlueprint;
+use re_viewer_context::ViewerContext;
+use re_viewport::{SpaceViewBlueprint, Viewport};
+
+#[derive(Default)]
+pub struct AddSpaceViewOrContainerModal {
+    target_container: Option<egui_tiles::TileId>,
+    modal_handler: re_ui::modal::ModalHandler,
+}
+
+impl AddSpaceViewOrContainerModal {
+    pub fn open(&mut self, target_container: egui_tiles::TileId) {
+        self.target_container = Some(target_container);
+        self.modal_handler.open();
+    }
+
+    pub fn ui(&mut self, ui: &mut egui::Ui, ctx: &ViewerContext<'_>, viewport: &Viewport<'_, '_>) {
+        self.modal_handler.ui(
+            ctx.re_ui,
+            ui,
+            || re_ui::modal::Modal::new("Add Space View or Container"),
+            |_, ui, _| modal_ui(ui, ctx, viewport, self.target_container),
+        );
+    }
+}
+
+fn modal_ui(
+    ui: &mut egui::Ui,
+    ctx: &ViewerContext<'_>,
+    viewport: &Viewport<'_, '_>,
+    target_container: Option<egui_tiles::TileId>,
+) {
+    ui.spacing_mut().item_spacing = egui::vec2(14.0, 10.0);
+
+    let container_data = [
+        (
+            "Tabs",
+            "Create a new tabbed container.",
+            egui_tiles::ContainerKind::Tabs,
+        ),
+        (
+            "Horizontal",
+            "Create a new horizontal container.",
+            egui_tiles::ContainerKind::Horizontal,
+        ),
+        (
+            "Vertical",
+            "Create a new vertical container.",
+            egui_tiles::ContainerKind::Vertical,
+        ),
+        (
+            "Grid",
+            "Create a new grid container.",
+            egui_tiles::ContainerKind::Grid,
+        ),
+    ];
+
+    for (title, subtitle, kind) in container_data {
+        if row_ui(ui, &re_ui::icons::CONTAINER, title, subtitle).clicked() {
+            viewport.blueprint.add_container(kind, target_container);
+            viewport.blueprint.mark_user_interaction(ctx);
+        }
+    }
+
+    ui.separator();
+
+    // space view of any kind
+    for space_view in ctx
+        .space_view_class_registry
+        .iter_registry()
+        .sorted_by_key(|entry| entry.class.display_name())
+        .map(|entry| {
+            SpaceViewBlueprint::new(
+                entry.class.identifier(),
+                &format!("empty {}", entry.class.display_name()),
+                &EntityPath::root(),
+                DataQueryBlueprint::new(entry.class.identifier(), EntityPathFilter::default()),
+            )
+        })
+    {
+        let icon = space_view.class(ctx.space_view_class_registry).icon();
+        let title = space_view
+            .class(ctx.space_view_class_registry)
+            .display_name();
+        let subtitle = format!("Create a new Space View to display {title} content.");
+
+        if row_ui(ui, icon, title, &subtitle).clicked() {
```

Comment:
Would be nice if this made the entire row clickable -- the "+" button hit target seems unnecessary in this context.